src/common/transformations/include/transformations/rt_info/reduceop_path.hpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/runtime_attribute.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+
+TRANSFORMATIONS_API void mark_reduceop_path(const std::shared_ptr<Node>& node);
+TRANSFORMATIONS_API bool is_reduceop_path(const std::shared_ptr<const Node>& node);
+
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief ReduceOpPath class represents runtime info attribute that marks path that goes
+ * into input of ReduceSum and ReduceMean. It is used to mark paths that goes from Exp to ReduceSum and ReduceMean
+ */
+class TRANSFORMATIONS_API ReduceOpPath : public RuntimeAttribute {
```

Comment:
do we really need `TRANSFORMATIONS_API`? Could you please check because it is better to minimize export symbols in the library?

Reply:
Frankly speaking this rt_info attribute and functions in `reduceop_path.hpp` meant to be used only internally. 

As far as i understand other marking in this folder `src/common/transformations/include/transformations/rt_info/` are also for internal usage only but they still are marked as `TRANSFORMATIONS_API`. I added this `TRANSFORMATIONS_API` just in analogy without thinking deeply.

@ilyachur can you help us? Do we need this `TRANSFORMATIONS_API` for runtime info attrs which we use only internally?

Reply:
As far as i understand, all rt_info attrs are placed in this king of directory and are marked with `TRANSFORMATION_API`.

A long time ago i introduced internal rt_info for nms->gather path and Gleb moved it to public space despite the fact it was totally internal. So I don't know what is right define them publicly or inside used transformation
https://github.com/openvinotoolkit/openvino/pull/7603/files#diff-8756c2d754bfe9081bdc3574086d8236a00b088a4ca3f981e8e9a5f303d63b93

@ilyachur @itikhono can you please help us to resolve this question? Where private rt_info definition should be placed and do we have to use `TRANSFORMATION_API`?

Reply:
Do we clear runtime info after using this attribute?
if not, it looks like this is not an internal attribute right now, runtime info is a part of ov Nodes and in case of we copy ov::Model, I expect all runtime attributes will be copied too. To copy this attribute object, copy ctor will be invoked, so most probably we will have an issue to call copy ctor (need to check this)

Reply:
After using no. Because other transformations do not use it. I'll check it's behavior during copying.

Do i understand correctly that because of this this class indeed must be public?
https://github.com/openvinotoolkit/openvino/blob/59c868f5ae28602c796f94b5a63b9f87f66147c6/src/core/include/openvino/core/node.hpp#L408-L413

Reply:
Interesting thing, copy constructor for `ov::Model` is deleted :)
https://github.com/openvinotoolkit/openvino/blob/0c9abf43a91fbe2dc95974b6cb2651bd883d1b12/src/core/include/openvino/core/model.hpp#L437-L440

@rkazants let's keep this attribute `ReduceOpPath` as is? It's exactly as other attributes also used only internaly. If we reconsider using `TRANSFORMATION_API` we should reconsider for all rt_info attrs and this is a separate task beyond scope of this PR

Reply:
You can copy model by call `model->clone()`.

Moreover why do you need this attribute in API? We use it only in one file...
Moreover can we use bool flag to mark nodes instead of introduction of new class?

Reply:
Since the rt_info attr is internal now and is removed after usage whether is cloneable does not matter.
But thanks for letting know about this method! I will know for future